scipy/optimize/_dual_annealing.py

```diff
@@ -38,9 +38,16 @@ class VisitingDistribution:
         makes the algorithm jump to a more distant region.
         The value range is (1, 3]. It's value is fixed for the life of the
         object.
-    rand_gen : {`~numpy.random.RandomState`, `~numpy.random.Generator`}
-        A `~numpy.random.RandomState`, `~numpy.random.Generator` object
-        for using the current state of the created random generator container.
+    rand_gen : {None, int, `numpy.random.Generator`,
+                `numpy.random.RandomState`}, optional
+
```

Comment:
This argument is not coerced via `check_random_state()`. This class and and `StrategyChain` are used internally after the `seed=` argument has been coerced already.

Reply:
OK, I though of doing it this way because `check_random_state` would still feed a `RandomState` to this class. I will update it then.

Reply:
Sorry, let me clarify: the old version of this docstring was correct and should just be reverted. The value is either a `RandomState` or a `Generator`, but it cannot be `None` or `int`. `VisitingDistribution` and `StrategyChain` are internal classes that will only be called with an actual `RandomState` or `Generator` instance. The user will call `dual_annealing()` with a `seed` argument; `dual_annealing()` is responsible for calling `rand_gen = check_random_state(seed)` and will then pass `rand_gen` (which is then only either a `RandomState` or `Generator` instance) to `VisitingDistribution` and `StrategyChain`.

Reply:
Oh right, I fixed it.